### PR TITLE
Fix for shared handles for pool deleter

### PR DESCRIPTION
**Pull request by Agrael1**

### CODE REVIEW

**Comment by asuessenbach** [comment]
I think, just using that check could result in a resource leak, as allocated command buffers would not be freed.
That is, I would prefer to have some template magic to resolve that

Besides that: You might have modified your vk.xml, somehow. the <commands> section is required and definitely part of every vk.xml. In the current version (1.3.291), it's on line 11321.

And a side question: is it somehow assured or verified, that the CommandPool passed into the SharedCommandBuffer constructor is the very same that is used to actually allocate the CommandBuffer?

**Comment by Agrael1** [comment]
> I think, just using that check could result in a resource leak, as allocated command buffers would not be freed. That is, I would prefer to have some template magic to resolve that
Template magic it is then.
> 
> And a side question: is it somehow assured or verified, that the CommandPool passed into the SharedCommandBuffer constructor is the very same that is used to actually allocate the CommandBuffer?
No way unfortunately, only if we had functions to make shared handles. In such case there would need to be all of the functions for shared handles. I dropped that idea, because both approaches looked inconvenient and ugly (operator-> and raw functions)

**Comment by Agrael1** [comment]
Ok, so I added the magic, should not compile without the explicit pool argument now

**Review by asuessenbach — COMMENTED** [review]
Looks good, so far.
Just two questions.

File: vulkan/vulkan_shared.hpp [hunk 1]
```diff
@@ -409,8 +444,8 @@ namespace VULKAN_HPP_NAMESPACE
   public:
     void destroy( DestructorType parent, HandleType handle ) const VULKAN_HPP_NOEXCEPT
     {
-      VULKAN_HPP_ASSERT( m_destroy && m_dispatch );
-      ( parent.*m_destroy )( m_pool.get(), 1u, &handle, *m_dispatch );
+      if ( m_destroy && m_dispatch && m_pool )
```
[the remark this answers is not in the document]
Comment (Agrael1) [reply vulkan/vulkan_shared.hpp:447]: hmm, I guess assert may be returned

File: vulkan/vulkansc_shared.hpp [hunk 2]
```diff
@@ -346,8 +346,8 @@ namespace VULKAN_HPP_NAMESPACE
   public:
     void destroy( DestructorType parent, HandleType handle ) const VULKAN_HPP_NOEXCEPT
     {
-      VULKAN_HPP_ASSERT( m_destroy && m_dispatch );
-      ( parent.*m_destroy )( handle, m_allocationCallbacks, *m_dispatch );
+      if ( m_destroy && m_dispatch )
```
[the remark this answers is not in the document]
Comment (Agrael1) [reply vulkan/vulkansc_shared.hpp:349]: About that, what do you think about making github action for generating the code on PR merge to master?

[the remark this answers is not in the document]
Comment (Agrael1) [reply vulkan/vulkansc_shared.hpp]: Ah, I saw, there is one already :D

**Comment by Agrael1** [comment]
@asuessenbach So, what do you think, can it be merged?

**Comment by asuessenbach** [comment]
Looks good now!
Thanks a lot for your contribution.

**Pull request merged**